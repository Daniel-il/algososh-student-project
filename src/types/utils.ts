import { ElementStates } from "./element-states";

export interface LettersStep {
  // Текущий порядок символов
  letters: string[];
  // Индекс изменяемого символа слева
  // Символ справа вычисляется на его основе, поэтому его хранить отдельно нет нужды
  // Если нет индекса, значит, сейчас ничего не меняется
  index?: number;
  // Состояние: changing - символ будет меняться, modified - символ изменился
  state: ElementStates[];
  swappedIndexes?: number[];
}
export type TQueueItem = {
  value: string;
  color: ElementStates;
};
export interface QueueState {
  items: string[];
}
