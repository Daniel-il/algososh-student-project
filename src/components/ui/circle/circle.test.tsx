import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

import { Circle } from "./circle";

describe("Тест компонента Circle", () => {
  it("Элемент без буквы", () => {
    const tree = renderer.create(<Circle letter="" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Элемент с буквой", () => {
    const tree = renderer.create(<Circle letter="tst" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Элемент с head", () => {
    const tree = renderer.create(<Circle head="1" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Circle c реакт-элементом в head", () => {
    const tree = renderer.create(<Circle head={<Circle />} />);

    expect(tree).toMatchSnapshot();
  });

  it("Элемент с tail", () => {
    const tree = renderer.create(<Circle tail="1" />);

    expect(tree).toMatchSnapshot();
  });

  it("Circle с реактом-элементом в tail", () => {
    const tree = renderer.create(<Circle tail={<Circle />} />);

    expect(tree).toMatchSnapshot();
  });

  it("Элемент с index", () => {
    const tree = renderer.create(<Circle index={0} />);

    expect(tree).toMatchSnapshot();
  });

  it("Элемент с пропом isSmall === true", () => {
    const tree = renderer.create(<Circle isSmall={true} />);

    expect(tree).toMatchSnapshot();
  });

  it("Элемент в состоянии Default", () => {
    const tree = renderer.create(<Circle state={ElementStates.Default} />);

    expect(tree).toMatchSnapshot();
  });

  it("Элемент в состоянии Changing", () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing} />);

    expect(tree).toMatchSnapshot();
  });

  it("Элемент в состоянии Modified", () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified} />);

    expect(tree).toMatchSnapshot();
  });
});
