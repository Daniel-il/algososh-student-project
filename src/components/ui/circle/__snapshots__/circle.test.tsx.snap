// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Circle Circle c реакт-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="undefined"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Circle с реактом-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="undefined"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тест компонента Circle Элемент без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент в состоянии Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент в состоянии Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент в состоянии Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    1
  </div>
</div>
`;

exports[`Тест компонента Circle Элемент с буквой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      tst
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Circle Элемент с пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="undefined"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
